checkov/cloudformation/checks/resource/base_resource_value_check.py

```diff
@@ -32,7 +32,7 @@ def _filter_key_path(path: str) -> List[str]:
         :param path: valid JSONPath of an attribute
         :return: List of named attributes with respect to the input JSONPath order
         """
-        return [x for x in path.split("/") if not re.search(r"^\[?\d+\]?$", x)]
+        return [x for x in path.split("/") if not re.search(re.compile(r"^\[?\d+\]?$", x))]
```

Comment:
I would refactor the `re.compile` out of the list comprehension so it won't compile the regex each iteration